### PR TITLE
revert: feat(navigation-user): add stateful component tokens

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #7180 

## Summary

Reverts [635e116](https://github.com/Esri/calcite-design-system/pull/8594/commits/635e116070de05ba92e4d6e0afb72bd70a845132) from `epic/7180-component-tokens`

### CODE REVIEW

**Review by jcfranco — APPROVED** [review]

File: packages/calcite-components/src/components/navigation-user/navigation-user.scss [hunk 1]
```diff
@@ -30,24 +26,26 @@
...
+    border-color: var(--calcite-navigation-user-border-color, transparent);
     font-size: var(--calcite-font-size-md);
     line-height: var(--calcite-font-line-height-fixed-lg);
   }
 }
 
 :host(:hover) .button,
 :host(:focus) .button {
-  background-color: var(--calcite-navigation-user-background-color-hover, var(--calcite-color-foreground-2));
+  --calcite-navigation-user-background-color: var(--calcite-color-foreground-2);
```
Comment (alisonailea) [inline packages/calcite-components/src/components/navigation-user/navigation-user.scss:39]: this should be `background-color: var(--calcite-navigation-user-background-color, var(--calcite-color-foreground-2));`

File: packages/calcite-components/src/components/navigation-user/navigation-user.scss [hunk 2]
```diff
@@ -30,24 +26,26 @@
...
 
 :host(:focus) .button {
   @apply focus-inset;
 }
 
 :host(:active) .button {
-  background-color: var(--calcite-navigation-user-background-color-active, var(--calcite-color-foreground-3));
-  color: var(--calcite-navigation-user-text-color-active, var(--calcite-color-text-1));
+  --calcite-navigation-user-background-color: var(--calcite-color-foreground-3);
+  --calcite-navigation-user-text-color: var(--calcite-color-text-1);
```
Comment (alisonailea) [inline packages/calcite-components/src/components/navigation-user/navigation-user.scss:48]: these should be

```
background-color: var(--calcite-navigation-user-background-color, var(--calcite-color-foreground-3));
color: var(--calcite-navigation-user-text-color, var(--calcite-color-text-1));
```

File: packages/calcite-components/src/components/navigation-user/navigation-user.scss [hunk 3]
```diff
@@ -60,8 +58,8 @@ calcite-avatar ~ .text-container {
 
 :host([active]) .button {
   --calcite-ui-icon-color: var(--calcite-color-brand);
-  border-color: var(--calcite-navigation-user-active-border-color, var(--calcite-color-brand));
-  color: var(--calcite-navigation-user-active-text-color, var(--calcite-color-text-1));
+  --calcite-navigation-user-text-color: var(--calcite-color-text-1);
+  --calcite-navigation-user-border-color: var(--calcite-color-brand);
```
Comment (alisonailea) [inline packages/calcite-components/src/components/navigation-user/navigation-user.scss:62]: these should be
```
border-color: var(--calcite-navigation-user-border-color, var(--calcite-color-brand));
color: var(--calcite-navigation-user-text-color, var(--calcite-color-text-1));
```

**Review by alisonailea — CHANGES_REQUESTED** [review]
Can you update these so we don't override the component tokens, just provide them as a fallback. If you want to only set the style props once and reassign a variable you can also do something like...

```
.button {
  border-color: var(--calcite-navigation-user-border-color, var(--calcite-internal-navigation-user-border-color, transparent));
  color: var(--calcite-navigation-user-text-color, var(--calcite-internal-navigation-user-text-color, inherit));
  background-color: var(--calcite-navigation-user-background-color, var(--calcite-internal, navigation-user-background-color, transparent));
}

host(:hover),
:host(:focus) {
  .button {
     --calcite-internal-navigation-user-background-color: var(--calcite-color-foreground-2);
  }
} 
```

**Pull request merged**